2022/opmodes/JavaLM0AutoBlue.java

```diff
@@ -0,0 +1,179 @@
+other materials provided with the distribution.
+
+Neither the name of FTC Teams 25/5218 nor the names of their contributors may be used to
+endorse or promote products derived from this software without specific prior
+written permission.
+
+NO EXPRESS OR IMPLIED LICENSES TO ANY PARTY'S PATENT RIGHTS ARE GRANTED BY THIS
+LICENSE. THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR
+TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+package opmodes;
+
+import com.qualcomm.robotcore.eventloop.opmode.Autonomous;
+import com.qualcomm.robotcore.hardware.DcMotor;
+import com.qualcomm.robotcore.util.RobotLog;
+
+import org.firstinspires.ftc.robotcore.external.Telemetry;
+
+import team25core.DeadReckonPath;
+import team25core.DeadReckonTask;
+import team25core.FourWheelDirectDrivetrain;
+import team25core.MechanumGearedDrivetrain;
+import team25core.OneWheelDirectDrivetrain;
+import team25core.Robot;
+import team25core.RobotEvent;
+import team25core.SingleShotTimerTask;
+
+
+@Autonomous(name = "JavaLM0AutoBlue")
+//@Disabled
+public class JavaLM0AutoBlue extends Robot {
+
+    private DcMotor frontLeft;
+    private DcMotor frontRight;
+    private DcMotor backLeft;
+    private DcMotor backRight;
+
+    private DcMotor carouselMech;
+    private OneWheelDirectDrivetrain singleMotorDrivetrain;
+    private DcMotor liftMotor;
+    private DcMotor intakeMotor;
+
+    private FourWheelDirectDrivetrain drivetrain;
+
+    DeadReckonPath firstPath;
+    DeadReckonPath secondPath;
+    DeadReckonPath carouselPath;
+
+    /*
+     * The default event handler for the robot.
+     */
+    @Override
+    public void handleEvent(RobotEvent e)
+    {
+        /*
+         * Every time we complete a segment drop a note in the robot log.
+         */
+        if (e instanceof DeadReckonTask.DeadReckonEvent) {
+            RobotLog.i("Completed path segment %d", ((DeadReckonTask.DeadReckonEvent)e).segment_num);
+        }
+    }
+
+    public void spinCarousel()
+    {
+        this.addTask(new DeadReckonTask(this, carouselPath, singleMotorDrivetrain){
+            @Override
+            public void handleEvent (RobotEvent e){
+                DeadReckonEvent path = (DeadReckonEvent) e;
+                if (path.kind == EventKind.PATH_DONE)
+                {
```

Comment:
Cuddled braces on all if statements please.  Please scan the entire diff for this.